src/HDF5Msg.chpl

```diff
@@ -951,19 +883,22 @@ module HDF5Msg {
         }
     }
 
-    proc writeLocalSegArray(file_id: C_HDF5.hid_t, group: string, vals, valDim: int, segs, segDim: int, type t) throws {
+    proc writeSegmentedLocalDset(file_id: C_HDF5.hid_t, group: string, vals, segs, write_offsets: bool, type t) throws {
         h5Logger.debug(getModuleName(),getRoutineName(),getLineNumber(),
                             "Writing group, %s".format(group));
 
         var dtype_id: C_HDF5.hid_t = getDataType(t);
-
-        var vd = valDim:C_HDF5.hsize_t;
-        C_HDF5.H5LTmake_dataset(file_id, "/%s/%s".format(group, SEGMENTED_VALUE_NAME).c_str(), 1:c_int, vd, dtype_id, vals.ptr);
+        var localVals: [0..#vals.size] t = vals.a;
+        var vd = vals.size:C_HDF5.hsize_t;
+        C_HDF5.H5LTmake_dataset(file_id, "%s/%s".format(group, SEGMENTED_VALUE_NAME).c_str(), 1:c_int, vd, dtype_id, c_ptrTo(localVals));
         writeArkoudaMetaData(file_id, "%s/%s".format(group, SEGMENTED_VALUE_NAME), "pdarray", dtype_id);
 
-        var sd = segDim:C_HDF5.hsize_t;
-        C_HDF5.H5LTmake_dataset(file_id, "/%s/%s".format(group, SEGMENTED_OFFSET_NAME).c_str(), 1:c_int, sd, getDataType(int), segs.ptr);
-        writeArkoudaMetaData(file_id, "%s/%s".format(group, SEGMENTED_OFFSET_NAME), "pdarray", getDataType(int));
+        if write_offsets {
+            var localSegs: [0..#segs.size] int = segs.a;
+            var sd = segs.size:C_HDF5.hsize_t;
+            C_HDF5.H5LTmake_dataset(file_id, "%s/%s".format(group, SEGMENTED_OFFSET_NAME).c_str(), 1:c_int, sd, getDataType(int), c_ptrTo(localSegs));
+            writeArkoudaMetaData(file_id, "%s/%s".format(group, SEGMENTED_OFFSET_NAME), "pdarray", getDataType(int));
```

Comment:
just want to verify that using `SEGMENTED_OFFSET_NAME` and `"pdarray"` won't be an issue in the segstr/categorical case

Reply:
It will not this allows the dataset to be read as a standalone pdarray. This was a feature requested by users.